lib/buffer.js

```diff
@@ -107,7 +111,7 @@ function Buffer(arg, encodingOrOffset, length) {
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    return allocate(arg);
```

Comment:
I mean, in that case it doesn't matter if I change this from `Buffer.allocUnsafe` to `allocate` or not? (your reasoning above seemed to be that it wouldn't throw on negative numbers if I keep as-is)